File: 1Urte desberdinetako materiala/Lorea 25-26/Android/Erronka-1/multi_intefaz/src/main/java/model/Liburu.java

package model;

import java.util.List;


public class Liburu {
    private int id;
    private String izenburua;
    private String idazle_izena;
    private List<String> generoak;

    public Liburu(int id, String izenburua, String idazle_izena, List<String> generoak) {
        this.id = id;
        this.izenburua = izenburua;
        this.idazle_izena = idazle_izena;
        this.generoak = generoak;
    }

    public Liburu(int id, String izenburua,List<String> generoak) {
        this.id = id;
        this.izenburua = izenburua;
        this.generoak = generoak;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getIzenburua() {
        return izenburua;
    }

    public void setIzenburua(String izenburua) {
        this.izenburua = izenburua;
    }

    public String getIdazlea() {
        return idazle_izena;
    }

    public void setIdazlea(String idazlea) {
        this.idazle_izena = idazlea;
    }

    public List<String> getGeneroak() {
        return generoak;
    }

    public void setGeneroak(List<String> generoak) {
        this.generoak = generoak;
    }
    
    // Devuelve la lista de géneros en formato cadena
    public String getGeneroakAsString() {
        return String.join(", ", generoak);
    }
  
    // Opcional: para mostrar el título en el TableView o ComboBox
    @Override
    public String toString() {
        return "LiburuDatu{" +
           "id=" + id +
           ", izenburua='" + izenburua + '\'' +
           ", generoak=" + generoak +
           ", idazle_izena='" + idazle_izena + '\'' +
           '}';
    }
}

File: PMD 25-26/Erronka/app/src/main/java/com/example/erronka/model/Liburu.kt
package com.example.erronka.model


/**class Liburu {
    var id: Int
    var izenburua: String?
    var idazlea: String? = null
    var generoak: MutableList<String?>

    constructor(
        id: Int,
        izenburua: String?,
        idazle_izena: String?,
        generoak: MutableList<String?>
    ) {
        this.id = id
        this.izenburua = izenburua
        this.idazlea = idazle_izena
        this.generoak = generoak
    }

    constructor(id: Int, izenburua: String?, generoak: MutableList<String?>) {
        this.id = id
        this.izenburua = izenburua
        this.generoak = generoak
    }

    val generoakAsString: String
        // Devuelve la lista de géneros en formato cadena
        get() = java.lang.String.join(", ", generoak)

    // Opcional: para mostrar el título en el TableView o ComboBox
    override fun toString(): String {
        return "LiburuDatu{" +
                "id=" + id +
                ", izenburua='" + izenburua + '\'' +
                ", generoak=" + generoak +
                ", idazle_izena='" + this.idazlea + '\'' +
                '}'
    }
}**/
class Liburu {

    private var id: Int = 0
    private var izenburua: String = ""
    private var idazle_izena: String = ""
    //private var generoak: List<String> = emptyList()
    private var generoak: MutableList<String> = mutableListOf()

    // Constructor completo
    //constructor(id: Int, izenburua: String, idazle_izena: String, generoak: List<String>) {
    constructor(id: Int, izenburua: String, idazle_izena: String, generoak: MutableList<String>) {
        this.id = id
        this.izenburua = izenburua
        this.idazle_izena = idazle_izena
        this.generoak = generoak
    }

    // Constructor sin idazle_izena
    // constructor(id: Int, izenburua: String, generoak: List<String>) {
    constructor(id: Int, izenburua: String, generoak: MutableList<String>) {
        this.id = id
        this.izenburua = izenburua
        this.generoak = generoak
    }

    // Getters y Setters
    fun getId(): Int = id
    fun setId(id: Int) { this.id = id }

    fun getIzenburua(): String = izenburua
    fun setIzenburua(izenburua: String) { this.izenburua = izenburua }

    fun getIdazlea(): String = idazle_izena
    fun setIdazlea(idazlea: String) { this.idazle_izena = idazlea }

    //fun getGeneroak(): List<String> = generoak
    //fun setGeneroak(generoak: List<String>) { this.generoak = generoak }

    // Getter público: devuelve MutableList para poder usar add()
    fun getGeneroak(): MutableList<String> = generoak

    // Setter público: recibe List o MutableList y lo convierte internamente a MutableList
    fun setGeneroak(list: MutableList<String>) {
        generoak = list.toMutableList()
    }

    // Liburua gehitzeko metodoa
    fun addLiburu(generoa: String) {
        generoak.add(generoa)
    }

    // Devuelve la lista de géneros como cadena
    fun getGeneroakAsString(): String = generoak.joinToString(", ")

    // toString equivalente
    override fun toString(): String {
        return "LiburuDatu(id=$id, izenburua='$izenburua', generoak=$generoak, idazle_izena='$idazle_izena')"
    }
}